### PR TITLE
Migration to webpack v4 and more !! 

### PR DESCRIPTION
**Pull request by markozxuu**
## Main changes ❤️ 😸 

The new version of webpack offers 2 modes that are `development and production`. The production mode optimizes the bundle size to the maximum

**v3.6.0**
screen shot 2018-09-11 at 23 00 06

**v4.6.0**
screen shot 2018-09-16 at 14 03 19

- All the dependencies of babel are updated to the new stable version (`babel v7`)

- The update of the new eslint version, recommends `Disallow Controls Characters in Regular Expressions`. I solved it by improving the regular expression ^-^

> Control characters are special, invisible characters in the ASCII range 0-31. These characters are rarely used in JavaScript strings so a regular expression containing these characters is most likely a mistake. [more info](https://eslint.org/docs/2.0.0/rules/no-control-regex)

- The warning of circular dependency was removed

## Minor changes 😈 

- All dependencies were updated

## Observation 👀 

The packages `flow-babel-webpack-plugin` apparently causes some problems with the new version of webpack v4

I was working on this PR a few weeks ago, it would be great if you could review it 🙏 
@rauchg @leo @TooTallNate 

#1291

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -93,95 +93,96 @@
...
     "arg": "2.0.0",
     "arr-flatten": "1.1.0",
-    "async-retry": "1.1.3",
-    "async-sema": "1.2.0",
+    "async-retry": "1.2.3",
+    "async-sema": "2.1.2",
     "ava": "0.25.0",
-    "aws-sdk": "2.124.0",
-    "babel-eslint": "8.2.2",
+    "aws-sdk": "2.315.0",
```
Comment (timneutkens) [inline package.json:115]: Looks like all deps were upgraded instead of just the ones related to webpack

Comment (markozxuu) [reply package.json:115]: that's bad? :o

Comment (timneutkens) [reply package.json:115]: Not necessarily, but I'd separate upgrading dependencies from implementing webpack 4, as both can be fragile changes.

Comment (markozxuu) [reply package.json:115]: Check each dependency you were updating to see if there are important changes, but most solve problems from previous versions : )

Comment (markozxuu) [reply package.json:115]: What do you recommend me to do ? :(

**Comment by joecohens** [comment]
Hi @MAPESO thank you for this PR we would really like to upgrade to Webpack v4 😄! 

I see you upgraded a lot of dependencies and I know it's important to do it, but we normally try to upgrade dependencies piece by piece since major and sometimes minor bumps can bring big changes and introduce bugs. 

Do you mind updating this PR so we only focus on upgrading all related specific to Webpack? And maybe later on make other PR's to upgrade different dependencies?

**Comment by markozxuu** [comment]
@joecohens Of course !!! :D then you recommend me to make small PR

**Comment by markozxuu** [comment]
thanks for the feedback @timneutkens @joecohens ❤️ I'm going to close this PR and I'll make a new one

**Pull request closed without merging**